app/models/questionnaire.rb

```diff
@@ -98,6 +98,8 @@ def final_question?
   def next_button_text
     if summative?
       final_question? ? 'Finish test' : 'Save and continue'
+    elsif certificate?
```

Comment:
`certificate?` is always `nil` because `OpenStruct` accepts any method call. The `module_item.certificate?` isn't relevant here so these lines can be removed I think.

Reply:
Ok, this was in response to the earlier change. Where should I up date so the next button text to s "Finish". Is this no longer required?

Reply:
no, youare correct - I conflated the two.  It's removed.